www/js/intro.js

```diff
@@ -102,6 +111,33 @@ angular.module('emission.intro', ['emission.splash.startprefs',
     $state.go('root.main.heatmap');
   };
 
+  $scope.os = function() {
+    if (ionic.Platform.isIOS()) {
+      $scope.OS = 'ios';
+    }
+    else {
+      $scope.OS = 'android';
+    }
+  }
+
+  $scope.fetchItinerumSurvey = function(url) {
+     var data_dict = {
+       'user': {
+         'uuid': $window.device.uuid,
```

Comment:
This is wrong. This is not the e-mission uuid; we don't read the device uuid because I was worried about being accused of fingerprinting, and it won't work anyway properly now without a IDFA etc because of fingerprinting.
Let's not do this and freak people out.

An example of reading the e-mission uuid is here:
https://github.com/e-mission/e-mission-phone/blob/5d964cbef72732fe4f2e9987d39c32c61d536433/www/js/survey/launch.js#L16

We might want to (after the paper is done) refactor that code into a common location to make it easier to use without copy-pasting.

Reply:
I am having issues figuring out where userProfile is coming from? I understand that it would be undefined if I just tried to do: `userProfile.user_id['$uuid']` but userProfile seems to pop up out of mid air and I can not trace back to where it comes from.

Reply:
line 13 in `launch.js`?

Reply:
Yes, but it looks like that then comes from `CommHelper.getUser()` in line 12? I believe it is the return value from this. However when I tried that approach the return value was undefined, and I thought we were not using CommHelper?

Reply:
We were not using `CommHelper` for the aggregate calls because it will only make calls to the e-mission server. However, `CommHelper.getUser` should work. It calls `"/profile/get"`; are you sure you are calling this after the `/profile/create` call in `intro.js` (called from `CommHelper.registerUser()`)?

If not, you need to move it after that call.

Reply:
This may be an issue. When I attempt to call the function I can the error with an undefined attribute uuid. Would this be an inferred error if we are calling before the registerUser? I am currently trying to track down the order in which these processes are called.

Reply:
How can I determine if $scope.login or $scope.agree is called first? What calls the intro.js file entirely?

Reply:
The entire "intro" sequence is an `ion-slide-box` I can't easily find the documentation on `ion-slide-box`, but IIRC, if you call `next()`, it goes to the next slide.


so in summary.html (which is the first slide, for example) we call `next` when the button is clicked, so it goes to the next screen.

```
<ion-footer-bar class="no-bgColor" id="intro-footer">
  <button class="button button-block button-outline button-positive" ng-click="next()">
    GETTING STARTED
  </button>
```

In sensor_explanatation.html (in the middle), we call `agree` when the button is clicked

```
  <div class="col" style="margin-left: 5px">
      <button class="button button-block button-positive button-outline" ng-click="agree()" id="consent-button"><span translate>{{'.button-accept'}}</span></button>
  </div>
```

So that's where `agree` is called. And `agree` in turn calls `next` to go to the next screen.

```
  $scope.agree = function() {
    StartPrefs.markConsented().then(function(response) {
      $ionicHistory.clearHistory();
      if ($state.is('root.intro')) {
        $scope.next();
      } else {
        StartPrefs.loadPreferredScreen();
      }
    });
  };
```

As in angular1 in general, you need to link the HTML to the controller to understand what is getting called where.